sql/CMakeLists.txt

```diff
@@ -723,8 +723,11 @@ ELSE()
   SET(MYSQLD_SOURCE main.cc)
 ENDIF()
 
+LIST(APPEND MYSQLD_SOURCE partitioning/partition_base.cc)
```

Comment:
I know we don't support WIN32, but, at the other hand, we don't remove the correspondent WIN32 pieces. What we see here is that MYSQLD_SOURCE is set up by some logic, it's not primitive assignment. Potentially adding partition_base.cc to source list is the common step and does not depend on the logic above. So it would be reasonable for me to separate it from the logic. But I can fix if you insist.

Reply:
But is the MYSQL_SOURCE the right place? Why not SQL_SOURCE?

Reply:
SQL_SOURCE contains source files list to build libsql.a in 5.7 and libsql_main.a in 8.0. This are static libraries. If they are linked to mysqld binary and some symbols from the libraries are not used in the binary, the binary will not contain such symbols. So, all symbols from partition_base.cc will not be linked in mysqld binary. Including partition_base.cc into source list of mysqld binary is the way to share it's code among plugins. Otherwise it would be necessary to link static libsql[_main].a to each plugin which uses partition_base.cc symbols, what is not good.

Reply:
ack, then ok as-is